src/components/trace-element.vue

```diff
@@ -1,17 +1,20 @@
 <template>
   <div class="pexpr" :class="classObj" id="id">
-    <div v-if="labeled" class="self">
+    <div v-if="labeled" class="self" :class="selfClassObj">
       <trace-label :traceNode="traceNode" :minWidth="minWidth"
                    @hover="onHover" @unhover="onUnhover" @click="onClick"
                    @showContextMenu="onShowContextMenu" />
+      <semantics-result v-if="hasResult" :traceNode="traceNode"
+                        @styleUpdate="updateSelfStyle"/>
     </div>
     <div v-if="!isLeaf" ref="children"
          class="children" :class="{vbox: vbox}"
          :hidden="collapsed">
       <trace-element v-for="child in childrenToRender"
                      :id="child.id" :traceNode="child.traceNode" :context="child.context"
                      :currentLR="child.currentLR" :measureInputText="measureInputText"
-                     :isInVBox="child.isInVBox" :eventHandlers="eventHandlers">
+                     :isInVBox="child.isInVBox" :eventHandlers="eventHandlers"
+                     :showResult="hasResult">
```

Comment:
It's a little confusing, so I rename the `showResult` into `initallyHasResult`. It is actually indicating whether a node should have a result by the time it's created. It's necessary because we are doing the selective rendering, so not all the nodes are created by default. `hasResult` is the actual state of having a result container appended to the node, it's manually mutable as need to update it upon selecting/unselecting an operation.

I am a little confused about the extra guard you mentioned. Are you suggesting that we should add a guard for showing the semantics editors as a whole, or just the semantics results that attached to the nodes? If you're thinking about the semantics results, I am not sure if the extra guard is necessary, as users could actually unselected the operations to disable showing the semantics results.